File: pdu/javascript/sample_action_msgs/pdu_cdr_conv_FibonacciActionRequest.js
import { FibonacciActionRequest } from './pdu_jstype_FibonacciActionRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ActionRequestHeader } from '../hako_action_msgs/pdu_jstype_ActionRequestHeader.js';
import { PduActionRequestHeaderConverter } from '../hako_action_msgs/pdu_cdr_conv_ActionRequestHeader.js';
import { FibonacciGoal } from '../sample_action_msgs/pdu_jstype_FibonacciGoal.js';
import { PduFibonacciGoalConverter } from '../sample_action_msgs/pdu_cdr_conv_FibonacciGoal.js';


export class PduFibonacciActionRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { FibonacciActionRequest } src
     */
    static to_cdr_body(writer, src) {
        PduActionRequestHeaderConverter.to_cdr_body(writer, src.header);
        PduFibonacciGoalConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { FibonacciActionRequest } dst
     * @returns { FibonacciActionRequest }
     */
    static cdr_body_to_js(reader, dst) {
        PduActionRequestHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduFibonacciGoalConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { FibonacciActionRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { FibonacciActionRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new FibonacciActionRequest());
    }
}

File: pdu/python/sample_action_msgs/pdu_cdr_conv_FibonacciActionRequest.py
from .pdu_pytype_FibonacciActionRequest import FibonacciActionRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_action_msgs.pdu_cdr_conv_ActionRequestHeader import *
from ..sample_action_msgs.pdu_cdr_conv_FibonacciGoal import *



def py_to_cdr_body_FibonacciActionRequest(writer: CdrWriter, src: FibonacciActionRequest):
    py_to_cdr_body_ActionRequestHeader(writer, src.header)
    py_to_cdr_body_FibonacciGoal(writer, src.body)


def cdr_body_to_py_FibonacciActionRequest(reader: CdrReader, dst: FibonacciActionRequest):
    cdr_body_to_py_ActionRequestHeader(reader, dst.header)
    cdr_body_to_py_FibonacciGoal(reader, dst.body)
    return dst


def py_to_cdr_FibonacciActionRequest(src: FibonacciActionRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_FibonacciActionRequest(writer, src)
    return writer.bytes()


def cdr_to_py_FibonacciActionRequest(cdr_payload) -> FibonacciActionRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = FibonacciActionRequest()
    return cdr_body_to_py_FibonacciActionRequest(reader, dst)
